agent/xds/rbac.go

```diff
@@ -518,18 +568,22 @@ func ixnSourceMatches(tester, against structs.ServiceName) bool {
 		return false
 	}
 
-	matchesAP := tester.PartitionOrDefault() == against.PartitionOrDefault() || against.PartitionOrDefault() == structs.WildcardSpecifier
```

Comment:
I removed `|| against.PartitionOrDefault() == structs.WildcardSpecifier` which isn't applicable since we validate against it